libgoal/libgoal.go

```diff
@@ -935,78 +935,81 @@ func MakeDryrunState(client Client, txnOrStxn interface{}, other []transactions.
 	return v2.DryrunRequestFromGenerated(&gdr)
 }
 
-// MakeDryrunStateGenerated function creates DryrunRequest data structure and serializes it into a file
+// MakeDryrunStateGenerated function creates generatedV2.DryrunRequest data structure and serializes it into a file
 func MakeDryrunStateGenerated(client Client, txnOrStxn interface{}, other []transactions.SignedTxn, proto string) (dr generatedV2.DryrunRequest, err error) {
 	var txns []transactions.SignedTxn
-	var tx *transactions.Transaction
-	if txn, ok := txnOrStxn.(transactions.Transaction); ok {
-		tx = &txn
+	if txnOrStxn == nil {
+		// empty input do nothing
+	} else if txn, ok := txnOrStxn.(transactions.Transaction); ok {
 		txns = append(txns, transactions.SignedTxn{Txn: txn})
 	} else if stxn, ok := txnOrStxn.(transactions.SignedTxn); ok {
-		tx = &stxn.Txn
 		txns = append(txns, stxn)
 	} else {
 		err = fmt.Errorf("unsupported txn type")
 		return
 	}
+
 	txns = append(txns, other...)
 	for i := range txns {
 		enc := protocol.EncodeJSON(&txns[i])
 		dr.Txns = append(dr.Txns, enc)
 	}
 
-	if tx.Type == protocol.ApplicationCallTx {
```

Comment:
entire `if tx.Type == protocol.ApplicationCallTx {` block moved inside the loop without modifications